core/services/pipeline/task.http.go

```diff
@@ -71,6 +73,7 @@ func (t *HTTPTask) Run(ctx context.Context, lggr logger.Logger, vars Vars, input
 		// Interpolated variable URLs use restricted HTTP adapter by default
 		// You must set allowUnrestrictedNetworkAccess=true on the task to enable variable-interpolated URLs to make restricted network requests
 		errors.Wrap(ResolveParam(&allowUnrestrictedNetworkAccess, From(NonemptyString(t.AllowUnrestrictedNetworkAccess), !variableRegexp.MatchString(t.URL))), "allowUnrestrictedNetworkAccess"),
+		errors.Wrap(ResolveParam(&reqHeaders, From(NonemptyString(t.Headers), "[]")), "reqHeaders"),
```

Comment:
I suggest we verify here the number of strings in slice is an even number and fail the task; then inside `makeHTTPRequest` we panic rather than logging a warning.